### PR TITLE
Refactor texture zero init tests and use WGSL

### PR DESCRIPTION
**Pull request by austinEng**
- Use WGSL
- Remove base and derived test classes. There is one test class
  which imports implementations of the read method
- Reorder parameters to make more sense from a test suppression point-of-view
- Handling of texture formats with extensions
- Parameter for with/without canary values
- More param filtering for invalid cases
- Various changes for concision and readability

### CODE REVIEW

**Comment by austinEng** [comment]
https://github.com/gpuweb/cts/compare/glsl-dependent...austinEng:zero-init-2

**Comment by austinEng** [comment]
Sadly looks like I changed too much stuff for it to diff nicely.

Related to #352 . I think the way I'm doing the data values there is better. Going from data rep -> shader rep instead of the other way around in these zero init tests.
Main goal here is to port this to the `main` branch where the other texelData stuff is happening so I can then get rid of the funky indirection from abstract "state" value to texel component value. Open to suggestions for further simplifications.

**Comment by Kangz** [comment]
+1, the way to do data rep is better in #352. Should we wait for that to land + this PR be rebased on top of it before reviewing?

**Comment by austinEng** [comment]
I think you could review now as the conversion to WGSL and parameterization changes are orthogonal to inverting the way data rep works, so I won't do all of that in this PR.

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (kainino0x) [inline src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:447]: nit: this could be simplified as below. Alternatively, could merge the `.filter(({ readMethod, aspect }) => ...` logic in here I think.

```suggestion
    if (info.depth && info.stencil) {
      return [{ aspect: 'depth-only' }, { aspect: 'stencil-only' }];
    }
    return [{ aspect: 'all' }];
```
or
```suggestion
    return poptions('aspect', (info.depth && info.stencil) ? ['depth-only', 'stencil-only'] : ['all']);
```

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (kainino0x) [inline src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:454]: Nit: these two `.unless`s only depend on readMethod and format, so can come immediately after both have been defined, near the top.

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (kainino0x) [inline src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:473]: Same, can be just after aspect is defined.

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (kainino0x) [inline src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:446]: This loses separate coverage for 'depth-only' and 'all' when both would be valid (same for stencil). Should we keep that coverage? It might behave a bit differently since implementations could have an error where they treat them differently for the purpose of identifying which aspects need to be cleared.

**Review by kainino0x — APPROVED** [review]
Shaders and params LGTM. Should we merge this first or 352?

File: src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts:5]: nit: function name doesn't have a verb in it.

File: src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts:24]: nit: I don't think this would be readback because it doesn't sample the texture. `getDepthTestEqualPipeline`?

File: src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts:120]: Shouldn't this be using a helper function to do the max(1, ...) ? ditto in other files.

File: src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/check_texture/by_ds_test.ts:65]: ditto

File: src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts:42]: Can we have a small example of the string that this produces? Is it a swizzle expression? (the variable could be renamed extraSwizzle?). Ah never mind we need the index expression for `vec2<f32>` but not `f32`. Can we have the comment that shows the code still?

File: src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts:111]: don't we want to check each mipLevel independently? Right now I believe only the first one will be checked.

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (Kangz) [inline src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:105]: I think this function could use the textureComponentType of the format instead of matching on substring?

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (Kangz) [reply src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:454]: Maybe this would help make the parameter generation marginally faster too?

**Review by Kangz — APPROVED** [review]

File: src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts
Comment (austinEng) [reply src/webgpu/api/operation/resource_init/check_texture/by_sampling.ts:111]: the level is passed in as a uniform

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:105]: It could, but that's in the other patch. I'll fix this up after landing both.

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (austinEng) [reply src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:446]: We should indeed keep that coverage. I'll add it back.

**Comment by austinEng** [comment]
@kainino0x PTAL again - I think this can land either before or after #352 - it shouldn't matter

File: src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts
Comment (kainino0x) [inline src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts:462]: This big params builder is a good candidate for subcases when we add them.

**Review by kainino0x — APPROVED** [review]

**Pull request merged**
